test/codeu/chat/server/JSONTest.java

```diff
@@ -0,0 +1,204 @@
+package codeu.chat.server;
+import static org.junit.Assert.*;
+
+import codeu.chat.common.ConversationHeader;
+import codeu.chat.common.Message;
+import codeu.chat.common.User;
+import codeu.chat.util.Uuid;
+import org.junit.Ignore;
+import org.junit.Test;
+import static org.junit.Assert.*;
+import java.io.IOException;
+import codeu.chat.util.store.Store;
+import java.lang.Iterable;
+import java.util.Iterator;
+import codeu.chat.util.Time;
+
+/**
+ * This class tests the JSON class in the server folder
+ * The purpose of this test is to ensure that the user is able to
+ * access the transaction log and that the transaction log is
+ * being saved properly.
+ */
+public class JSONTest {
+
+    @Test
+    public void readTest1() {
+        // This function tests if the JSON reader is working if there is a JSON with one type of each Object in the file
+        Model model = new Model();
+        JSON log = new JSON();
+        Model comparer = new Model();
+        comparer.add(new User(createUuid("1.1074501217"), "Krager", Time.fromMs(4309509)));
+        comparer.add(new ConversationHeader(createUuid("1.3959833157"), createUuid("1.1074501217"),  Time.fromMs(59408509), "Chat1"));
+        comparer.add(new Message(createUuid("1.3928689216"), Uuid.NULL, Uuid.NULL, Time.fromMs(49584908), createUuid("1.1074501217"),   "I'm so alone"));
+
+        try {
+            // Path used by Priyanka:
+            model = log.readFromFile("/Users/HMCLoaner/Desktop/CodeU/test/codeu/chat/server/input.json");
+            // The path can vary from device, so this may not compile if it's not the correct path for your project
+            // set up. A suggested path is as follows
+            // model = log.readFromFile("CodeU/test/codeu/chat/server/input.json");
+        } catch (IOException e) {
+            System.out.println("File not handled properly");
+        }
+        assertEquals(
+                compare(model.userById().all().iterator(), comparer.userById().all().iterator()), true);
```

Comment:
Is there a way to just compare the collections directly instead of going to the iterator? What does `model.userById()` return?